querydsl-jpa/src/test/java/com/mysema/query/AbstractJPATest.java

```diff
@@ -547,6 +547,26 @@ public void DistinctResults() {
     }
 
     @Test
+    public void Distinct_OrderBy() {
+        QCat cat = QCat.cat;
+        assertEquals(6, query()
+                .distinct()
+                .from(cat)
+                .orderBy(cat.mate.id.asc())
+                .list(cat.id, cat.mate.id).size());
+    }
+
+    @Test
+    public void Distinct_OrderBy2() {
+        QCat cat = QCat.cat;
+        assertEquals(4, query()
```

Comment:
Also, wouldn't you want to assert something about the order or the uniqueness of the elements instead of looking at how many are returned?